The rushed and apparently not widely tested EXA code does not work
on MGA Millenium 2 for me.

diff --git a/src/mga_driver.c b/src/mga_driver.c
index be0b3bc..441ba65 100644
--- a/src/mga_driver.c
+++ b/src/mga_driver.c
@@ -1957,7 +1957,8 @@ MGAPreInit(ScrnInfoPtr pScrn, int flags)
 	pMga->NoAccel = FALSE;
 	pMga->Exa = FALSE;
 #ifdef USE_EXA
-	pMga->Exa = TRUE;
+	if (pMga->Chipset > PCI_CHIP_MGA2164_AGP)
+		pMga->Exa = TRUE;
 	if (!xf86NameCmp(s, "EXA")) {
 	    pMga->Exa = TRUE;
 	    from = X_CONFIG;
